src/App.tsx

```diff
@@ -2,30 +2,74 @@ import React from 'react';
 import { Container, Stack } from '@mui/material';
 
 import { EditDialog } from 'modals';
-import { AddTask, TasksList } from 'components/tasks';
-import { Task } from 'types/task';
+import { AddTaskForm, TasksList } from 'components/tasks';
+import { Task, TaskStatus } from 'types/task';
 
 export const App: React.FC = () => {
   const [openEditDialog, setOpenEditDialog] = React.useState(false);
   const [tasks, setTasks] = React.useState<Task[]>([]);
-  const [editTask, setEditTask] = React.useState<Task>({} as Task);
+  const [editTask, setEditTask] = React.useState<Task | null>(null);
+
+  const onTaskStatusChange = (taskId: number): void => {
+    const newTasks = tasks.map((task: Task) => {
+      if (task.id === taskId) {
+        task.status =
+          task.status === TaskStatus.Created ? TaskStatus.Started : TaskStatus.Completed;
+      }
+      return task;
+    });
+    setTasks(newTasks);
+  };
+
+  const onCreateTask = (taskName: string): void => {
+    const task: Task = {
+      id: Date.now(),
+      name: taskName,
+      date: new Date(),
+      status: TaskStatus.Created,
+    };
+    setTasks([task, ...tasks]);
+  };
+
+  const onDeleteTask = (taskId: number): void => {
+    setTasks(tasks.filter(t => t.id !== taskId));
+  };
+
+  const onEditTask = (task: Task): void => {
+    setEditTask(task);
+    setOpenEditDialog(true);
+  };
+
+  const onEditTaskDialogSave = (editedTask: Task): void => {
+    setOpenEditDialog(false);
+    const newTasks = tasks.map((task: Task) => {
+      if (task.id === editedTask.id) {
+        task = editedTask;
+      }
+      return task;
+    });
+    setTasks(newTasks);
```

Comment:
```suggestion
    const newTasks = tasks.map((task: Task) => {
      if (task.id === editedTask.id) {
        return editedTask
      }
  
      return task;
    });

    setTasks(newTasks);
    setOpenEditDialog(false);
```
сначала работаем с данными, сохраняем их в стейт и если все ок, то закрываем модалку
может произойти ошибка во время редактирования или сетевая ошибка, а модалка уже закрыта